src/framework/midi/internal/platform/lin/alsamidiinport.h

```diff
@@ -35,9 +35,10 @@ class AlsaMidiInPort : public IMidiInPort, public async::Asyncable
 {
 public:
     AlsaMidiInPort() = default;
-    ~AlsaMidiInPort() override;
+    ~AlsaMidiInPort() = default;
```

Comment:
I'm not quite sure whether this one is correct. The reason I'm doubting is that we have a member `std::shared_ptr<Alsa> m_alsa;`, and the Alsa struct is defined only in the cpp file. I would expect that `shared_ptr` needs to know the definition of Alsa in its destructor (because it might need to delete the underlying pointer, and therefore it needs to know the size of the Alsa struct). Therefore I'm thinking that the destructor of AlsaMidiInPort etc. needs to be defined in the cpp file, so that the destructor of m_alsa is (implicitly) done in cpp where the side of Alsa is known. But for some reason it still compiles like this, so maybe the destructor of shared_ptr does something else than I'm thinking?

Edit: seems that's the case... apparently it does something with a "deleter", that is created when the shared_ptr is constructed. That explains a lot.